### ISSUE TITLE
Last argument is parsed as a filter rather than as a filename

### ISSUE THREAD

**Issue by cykl** [jbruggem/jaslog#1]
Printing a file works fine but as soon as I add a filter I get a panic. It seems that it tries to parse the last argument as a filter rather than a filename.

```
$ cat test
{ "level": "WARN", "logger": "root", "message": "Hello World!", "host": "vm", "@timestamp": "2013-11-17T10:21:41.863Z", "thread": "main", "@version": "1" }

$ RUST_BACKTRACE=full jaslog  -f host=vm test1 
Can't parse filter: test1
thread 'main' panicked at 'Error.', src/filter.rs:88:7
stack backtrace:
   0:     0x55ea3247fb54 - backtrace::backtrace::libunwind::trace::h65597d255cb1398b
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1:     0x55ea3247fb54 - backtrace::backtrace::trace_unsynchronized::hd4f479d7150ec4a0
                               at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2:     0x55ea3247fb54 - std::sys_common::backtrace::_print_fmt::h015072984a2b172c
                               at src/libstd/sys_common/backtrace.rs:77
   3:     0x55ea3247fb54 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h6df05d3335f32194
                               at src/libstd/sys_common/backtrace.rs:61
   4:     0x55ea3249f6dc - core::fmt::write::h1f444f4312eb6c27
                               at src/libcore/fmt/mod.rs:1028
   5:     0x55ea3247dd07 - std::io::Write::write_fmt::h8d147888220078ef
                               at src/libstd/io/mod.rs:1412
   6:     0x55ea32481f8e - std::sys_common::backtrace::_print::h8a6df0fa81d6af62
                               at src/libstd/sys_common/backtrace.rs:65
   7:     0x55ea32481f8e - std::sys_common::backtrace::print::h6f05b4733407e509
                               at src/libstd/sys_common/backtrace.rs:50
   8:     0x55ea32481f8e - std::panicking::default_hook::{{closure}}::h0d0a23bd02315dd8
                               at src/libstd/panicking.rs:188
   9:     0x55ea32481c81 - std::panicking::default_hook::h8d15a9aecb4efac6
                               at src/libstd/panicking.rs:205
  10:     0x55ea3248262b - std::panicking::rust_panic_with_hook::hbe174577402a475d
                               at src/libstd/panicking.rs:464
  11:     0x55ea32389185 - std::panicking::begin_panic::hd87a1c47675581ec
  12:     0x55ea32387eec - jaslog::filter::Filter::from::hf4ee6175d083155b
  13:     0x55ea323870ef - jaslog::filter::parse_filters::h11365ae155098cab
  14:     0x55ea3237f248 - jaslog::read_log::h4e3ca832e1afe79b
  15:     0x55ea3237d6c6 - jaslog::main::hd436022fad0a2614
  16:     0x55ea3237bd73 - std::rt::lang_start::{{closure}}::h249fd722cac3989b
  17:     0x55ea32482053 - std::rt::lang_start_internal::{{closure}}::h6ea535ec5c50fc3e
                               at src/libstd/rt.rs:48
  18:     0x55ea32482053 - std::panicking::try::do_call::h631c6408dfccc6f5
                               at src/libstd/panicking.rs:287
  19:     0x55ea324842aa - __rust_maybe_catch_panic
                               at src/libpanic_unwind/lib.rs:78
  20:     0x55ea32482b0d - std::panicking::try::hab539b2d1255d635
                               at src/libstd/panicking.rs:265
  21:     0x55ea32482b0d - std::panic::catch_unwind::hd5e0a26424bd7f34
                               at src/libstd/panic.rs:396
  22:     0x55ea32482b0d - std::rt::lang_start_internal::h3bdc4c7d98181bf9
                               at src/libstd/rt.rs:47
  23:     0x55ea3237d7c2 - main
  24:     0x7fcb6dbed1a3 - __libc_start_main
  25:     0x55ea3237a7ca - _start
  26:                0x0 - <unknown>
```

Environment:

 - jaslog 0.1.4
 - Fedora 31

**Comment by jbruggem** [comment]
Thanks for the report.

It's a known bug, and I *think* it's due to either a bug in [clap](https://github.com/clap-rs/clap), or a mistake in [how I use clap to parse arguments](https://github.com/jbruggem/jaslog/blob/master/src/main.rs#L26).

To avoid this error, make sure that the file name is always the first argument : 
```
jaslog test1 -f host=vm
```
Or pipe in jaslog:
```
cat test1 | jaslog -f host=vm
```

**Comment by cykl** [comment]
Thanks for the prompt answer! Reordering arguments in the README might help adoption.

**Comment by jbruggem** [comment]
Thanks ! 

I'll check the readme and I'll keep this open because I'd like to solve this at some point :).

**Comment by jbruggem** [comment]
The solution was in the documentation of clap:

https://docs.rs/clap/2.33.0/clap/struct.Arg.html#method.multiple

> WARNING:
> 
> Setting multiple(true) for an option with no other details, allows multiple values and multiple occurrences [...]. Because multiple values are allowed, --option val1 val2 val3 is perfectly valid, be careful when designing a CLI where positional arguments are expected after a option which accepts multiple values [...].
> 
> Pro Tip:
> 
> It's possible to define an option which allows multiple occurrences, but only one value per occurrence. To do this use Arg::number_of_values(1) in coordination with Arg::multiple(true).

**Comment by jbruggem** [comment]
Fixed in: [v0.1.5](https://github.com/jbruggem/jaslog/releases/tag/v0.1.5)
